packages/core/src/stateUtils.ts

```diff
@@ -1497,15 +1503,20 @@ export function resolveActionsAndContext<
   currentState: State<TContext, TEvent, any>,
   actorCtx: ActorContext<any, any> | undefined
 ): {
-  actions: typeof actions;
-  raised: Array<RaiseActionObject<TEvent>>;
-  context: TContext;
+  nextState: AnyState;
 } {
   const { machine } = currentState;
   const resolvedActions: BaseActionObject[] = [];
   const raiseActions: Array<RaiseActionObject<TEvent>> = [];
-  let { context } = currentState;
-  let updatedContext = context;
+  let intermediateState = currentState;
+
+  function handleAction(action: BaseActionObject): void {
+    resolvedActions.push(action);
+    if (actorCtx?.self.status === ActorStatus.Running) {
+      action.execute?.(actorCtx!);
+      delete action.execute;
```

Comment:
Hmm, kinda hacky... but let's roll with it for now

Reply:
Yeah. This really only applies to restoring a state & starting from an initial state where actions need to be executed.